test/coverage/utils/ExpectedTest.cpp

```diff
@@ -32,7 +74,126 @@ class TestFixture : public testing::Test {
 	static void TearDownTestSuite() {}
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+TEST_F(TestFixture, ExpectScalar) {
+	auto exp = parse_number("44");
+	EXPECT_EQ(true, bool(exp));
+	EXPECT_EQ(true, exp.has_value());
+	EXPECT_EQ(44, exp.value());
+	EXPECT_EQ(44, *exp);
+}
+
+TEST_F(TestFixture, ExpectComposite) {
```

Comment:
Some non-trivial types should also be tested (e.g. `std::string` or a struct/class with a destructor).